services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -784,7 +785,8 @@ public static void createLoginCookie(KeycloakSession keycloakSession, RealmModel
         CookieHelper.addCookie(KEYCLOAK_IDENTITY_COOKIE, encoded, cookiePath, null, null, maxAge, secureOnly, true, SameSiteAttributeValue.NONE);
         //builder.cookie(new NewCookie(cookieName, encoded, cookiePath, null, null, maxAge, secureOnly));// todo httponly , true);
 
-        String sessionCookieValue = realm.getName() + "/" + user.getId();
+        // With user-storage providers, user ID can contain special characters, which need to be encoded
+        String sessionCookieValue = realm.getName() + "/" + Encode.urlEncode(user.getId());
```

Comment:
Curious why the session cookie has the realm name, and user-id. Session iframe only needs the session-id afaik.

Reply:
I also don't know. Especially realm is in the cookie path and user-id does not seem to be needed for anything.

On the other hand, if we change it, we break compatibility with the keycloak-js adapter and we break zero-downtime upgrade in z-stream, right?  For example case when login is served by cluster node1, who will use cookie with only session-id, but admin console is served on cluster node2, which still expects old iframe format, then login to the admin console might not work as expected etc.

Hence should we do this one for z-stream and then have follow-up issue, which will change cookie to only session-id? That one will need to change iframe as well and will require migration instructions though. WDYT?